Dockerfile

```diff
@@ -77,17 +77,27 @@ apt install -y --no-install-recommends \
     lua5.4 \
     build-essential \
     ca-certificates \
-    wget
+    curl
 EOF
 
 WORKDIR /opt/build
 
+# install built-ins
+ARG BUILTINS_VERSION=0.4.0
+RUN <<EOF
+set -eu
+curl -L -R -O https://github.com/crypto-bug-hunters/builtins/releases/download/v${BUILTINS_VERSION}/builtins-${BUILTINS_VERSION}.tar.gz
+mkdir -p builtins
+tar -xf builtins-${BUILTINS_VERSION}.tar.gz -C builtins
+rm builtins-${BUILTINS_VERSION}.tar.gz
+EOF
+
```

Comment:
Now that we have builtins published as binaries and container registry we could remove this

```suggestion
```

And at the final stage use one of these options

**Using `ADD --checksum`**

```
ADD --checksum=f610ff16ff7948c8b26bcbbedc5f5b733cdd06496231ec699cdda2c5a01ee1c2 https://github.com/crypto-bug-hunters/builtins/releases/download/v0.5.0/busybox-1.36.1-linux-riscv64 /usr/bin/busybox-1.36.1
```

**Using `COPY --from`**

```
COPY --from=ghcr.io/crypto-bug-hunters/builtins:0.5.0 --chmod=755 /opt/bundle/busybox-1.36.1-linux-riscv64  /usr/bin/busybox-1.36.1
```

Reply:
Let's use `COPY --from` as it favors:

- Local prototyping (by tagging images accordingly)
- Mirroring (GHCR, DockerHub, etc.)